File: python/src/1752-check.py
# https://leetcode.cn/problems/check-if-array-is-sorted-and-rotated/
class Solution:
    def check(self, nums: List[int]) -> bool:
        tmp = 0
        nums_len = len(nums)
        if nums_len < 3:
            return True
        if nums[0] < nums[nums_len - 1]:
            tmp = tmp + 1
        for i in range(0, nums_len - 1):
            if nums[i] > nums[i + 1]:
                tmp = tmp + 1
            if tmp > 1:
                return False
        return True

File: typescript/src/1752-check.ts
function check(nums: number[]): boolean {
  const len: number = nums.length;
  // 长度小于3的数组肯定可以旋转成排序好的数组
  if (len < 3) {
    return true;
  }
  let add = 0;
  if (nums[len - 1] > nums[0]) {
    add++;
  }
  for (let i = 0; i < len - 1; i++) {
    if (nums[i] > nums[i + 1]) {
      add++;
    }
    if (add > 1) {
      return false;
    }
  }
  return true;
}

export {check};
